packages/components/focus-trap/src/focus-trap.vue

```diff
@@ -35,249 +35,248 @@ import {
 import type { PropType } from 'vue'
 import type { FocusLayer } from './utils'
 
-export default defineComponent({
+defineOptions({
   name: 'ElFocusTrap',
   inheritAttrs: false,
-  props: {
-    loop: Boolean,
-    trapped: Boolean,
-    focusTrapEl: Object as PropType<HTMLElement>,
-    focusStartEl: {
-      type: [Object, String] as PropType<'container' | 'first' | HTMLElement>,
-      default: 'first',
-    },
-  },
-  emits: [
-    ON_TRAP_FOCUS_EVT,
-    ON_RELEASE_FOCUS_EVT,
-    'focusin',
-    'focusout',
-    'focusout-prevented',
-    'release-requested',
-  ],
-  setup(props, { emit }) {
-    const forwardRef = ref<HTMLElement | undefined>()
-    let lastFocusBeforeTrapped: HTMLElement | null
-    let lastFocusAfterTrapped: HTMLElement | null
+})
 
-    useEscapeKeydown((event) => {
-      if (props.trapped && !focusLayer.paused) {
-        emit('release-requested', event)
-      }
-    })
+const props = defineProps({
```

Comment:
Please read this https://github.com/element-plus/element-plus/pull/3113

> 🧲 Extract .vue props and emits to .ts.
- Philosophy: Write the logical part into ts file as much as possible.
- add InstanceType for each components https://github.com/element-plus/element-plus/issues/4308